From 58b3f6fb090a646abbf4d361a89eb5a2526abcc5 Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Wed, 16 Sep 2015 09:31:46 +0530
Subject: [PATCH] FROM_UPSTREAM [VPG]: drm/i915: Remove HBR2 for CHT

HBR2 is not supported in CHT hence this patch removes it from
the maximum link rate supported for CHT.

This is modified version of the patch in upstream that performs the same
operation.

Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: 5e86dfe39f54ab13fd8079ac3d6cb100318909a3
Upstream-as-of-tag: drm-intel-next-2015-05-22-14655-g5e86dfe
Change-Id: I8c4b242503d4932a183b72a41c48a5345bc5aa6a
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-5971
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index f7d168b..a7d0d58 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -134,7 +134,7 @@ intel_dp_max_link_bw(struct intel_dp *intel_dp)
 		break;
 	case DP_LINK_BW_5_4: /* 1.2 capable displays may advertise higher bw */
 		if (((IS_HASWELL(dev) && !IS_HSW_ULX(dev)) ||
-		     INTEL_INFO(dev)->gen >= 8) &&
+		    IS_BROADWELL(dev) || (INTEL_INFO(dev)->gen >= 9)) &&
 		    intel_dp->dpcd[DP_DPCD_REV] >= 0x12)
 			max_link_bw = DP_LINK_BW_5_4;
 		else
-- 
1.9.1

